demo/lightning/checkpoint/simulated/multiprocessing_train.py

```diff
@@ -54,37 +150,39 @@ def save_checkpoint(self, checkpoint: Dict[str, torch.Tensor], filepath: str, st
         dist_cp.save(state_dict=checkpoint, checkpoint_id=filepath,
                      storage_writer=self.writer)
 
-    def load_checkpoint(self, checkpoint_path: str) -> None:
+    def load_checkpoint(self, checkpoint_path: str, initial_state_dict: Dict) -> None:
         """
         Loads a model's state dictionary from a specified checkpoint file in GCS. torch.distributed.checkpoint.load contains the core logic of loading sharded model weights.
         You can find the source code for FSDP.load_checkpoint
         https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/fabric/strategies/fsdp.py#L519.
 
+        For torch.distributed.checkpoint.load to work properly the template_state_dict should have model format.
```

Comment:
What does "model format" mean here? Match the dimensions of the model's input tensor?

Reply:
Correct.
Number of layers, size of tensors.